File: day-02/workspace/java-oop/src/main/java/com/targetindia/model/Triangle.java
package com.targetindia.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Triangle implements GeometricShape{
    private double base;
    private double height;

    @Override
    public double calculateArea() {
        return 0.5*base*height;
    }

    @Override
    public String getShapeName() {
        return "TRIANGLE";
    }
}

File: day-13-advanced-kotlin-concepts/workspace/kotlin-interfaces/src/main/kotlin/com/targetindia/model/Triangle.kt
package com.targetindia.model

class Triangle : IGeometricShape, IMovable {
    var x: Double = 0.0
    var y: Double = 0.0
    var base: Double = 1.0
    var height: Double = 1.0

    constructor()

    constructor(base: Double, height: Double) {
        this.base = base
        this.height = height
    }

    override fun getName(): String {
        return "Triangle"
    }

    override fun getArea(): Double {
        return 0.5 * base * height
    }

    override fun moveLeft(by: Double) {
        x -= by
    }

    override fun moveRight(by: Double) {
        x += by
    }

    override fun moveUp(by: Double) {
        y -= by
    }

    override fun moveDown(by: Double) {
        y += by
    }
}